xquant/keras/keras_report_utils.py

```diff
@@ -0,0 +1,380 @@
+        """
+        quant_points_names = [
+            layer.name for layer in quantized_model.layers
+            if isinstance(layer, KerasQuantizationWrapper)
+        ]
+
+        float_name2quant_name = {}
+
+        for quant_point in quant_points_names:
+            candidate_float_layer_name = quantized_model.get_layer(quant_point).layer.name
+
+            if candidate_float_layer_name in [layer.name for layer in float_model.layers]:
+                if candidate_float_layer_name not in float_name2quant_name:
+                    float_name2quant_name[candidate_float_layer_name] = quant_point
+                else:
+                    Logger.critical(f"Duplicate mapping found for layer: {candidate_float_layer_name}")
+            else:
+                Logger.warning(f"Skipping point {quant_point}")
+
+        return float_name2quant_name
+
+    def get_quantized_graph(self,
+                            quantized_model: Model,
+                            repr_dataset: Callable = None):
+        """
+        Get a graph representation of the quantized model.
+
+        Args:
+            quantized_model: The quantized model.
+            repr_dataset: Representative dataset to use during the graph building (not used in Keras).
+
+        Returns:
+            Graph representation of the quantized model.
+        """
+        quant_graph = model_reader(quantized_model)
+        return quant_graph
+
+
+    def get_quant_graph_with_metrics(self,
+                                     quantized_model: keras.Model,
+                                     collected_data: Dict[str, Any],
+                                     xquant_config: XQuantConfig,
+                                     repr_dataset: Callable = None) -> Graph:
+        """
+        Generate the quantized graph with associated metrics.
+
+        Args:
+            quantized_model (keras.Model): The quantized Keras model.
+            collected_data (Dict[str, Any]): Data collected during quantization.
+            xquant_config (XQuantConfig): Configuration settings for explainable quantization.
+            repr_dataset (Callable): Representative dataset (not used in Keras).
+
+        Returns:
+            Graph: A graph structure with metrics.
+        """
+        quant_graph = self.get_quantized_graph(quantized_model=quantized_model)
+        for node in quant_graph.nodes:
+            if xquant_config.compute_intermediate_metrics_repr:
+                if node.name in collected_data[INTERMEDIATE_METRICS_REPR].keys():
+                    node.framework_attr[XQUANT_REPR] = collected_data[INTERMEDIATE_METRICS_REPR][node.name]
+            if xquant_config.compute_intermediate_metrics_val:
+                if node.name in collected_data[INTERMEDIATE_METRICS_VAL].keys():
+                    node.framework_attr[XQUANT_VAL] = collected_data[INTERMEDIATE_METRICS_VAL][node.name]
+        return quant_graph
+
+    def add_histograms_to_tensorboard(self,
```

Comment:
just so we remember to talk about this, I think that this is another example for a function that needs to be split and part of it extracted from this module